src/main/java/com/puppycrawl/tools/checkstyle/checks/blocks/RightCurlyCheck.java

```diff
@@ -71,6 +71,50 @@
  * &lt;module name="RightCurly"/&gt;
  * </pre>
  * <p>
+ * Example:
+ * </p>
+ * <pre>
+ *public class Test {
+ *
+ *  public void test() {
+ *
+ *    if (foo) {
+ *      bar();
+ *    }           // violation, right curly must be in the same line
+ *    else {
+ *      bar();
+ *    }
+ *
+ *    if (foo) {
+ *      bar();
+ *    } else {     // OK
+ *      bar();
+ *    }
+ *
+ *    if (foo) { bar(); } int i = 0; // violation
+ *
+ *    if (foo) { bar(); }            // OK
+ *    int i = 0;
+ *
+ *    try {
+ *      bar();
+ *    }                       // violation
```

Comment:
please add brief comment on why violation is here, please do same for other cases.